sql/multi_range_read.cc

```diff
@@ -795,9 +795,11 @@ int Mrr_ordered_index_reader::init(handler *h_arg, RANGE_SEQ_IF *seq_funcs,
 }
 
 
-static int rowid_cmp_reverse(void *file, uchar *a, uchar *b)
+static int rowid_cmp_reverse(const void *file, const void *a, const void *b)
 {
-  return - ((handler*)file)->cmp_ref(a, b);
+  return -((handler *) file)
+              ->cmp_ref(static_cast<const uchar *>(a),
+                        static_cast<const uchar *>(b));
```

Comment:
Adding `const` to `handler::cmp_ref()` might be a bit tricky because of spider. Spider's `cmp_ref`:
 1. Modifies data (which we can at least tag with `mutable`), so this is less important, but also
 2. Calls `handler::rnd_pos()`, which (to my understanding) would require `rnd_pos()` to also be `const`, where other implementations of `rnd_pos` also modify member variables (e.g. partition). I'm not too familiar with that part of the code base, so I'll need to explore how deep that rabbit hole goes (unless you have another suggestion).

Reply:
I didn’t actually suggest any change to `handler::cmp_ref()`. I mainly observed that the C-style cast is casting away `const` from the `file` pointer target, and then potentially modifying that object. I think that it would be cleaner to not change the type signature of `qsort_cmp2` in the first place, that is, not to add the bogus `const`.